packages/language-chooser/src/language-chooser/index.tsx

```diff
@@ -0,0 +1,348 @@
+		setSelectedLanguage( locale );
+	};
+
+	const isEmpty = languages.length === 0;
+	const isMoveUpDisabled =
+		! selectedLanguage ||
+		languages[ 0 ]?.locale === selectedLanguage?.locale;
+	const isMoveDownDisabled =
+		! selectedLanguage ||
+		languages[ languages.length - 1 ]?.locale === selectedLanguage?.locale;
+	const isRemoveDisabled = ! selectedLanguage;
+
+	const onAdd = () => {
+		onAddLanguage( selectedInactiveLanguage );
+
+		const installedIndex = installedLanguages.findIndex(
+			( { locale } ) => locale === selectedInactiveLanguage.locale
+		);
+
+		const availableIndex = availableLanguages.findIndex(
+			( { locale } ) => locale === selectedInactiveLanguage.locale
+		);
+
+		let newSelected: Language | undefined;
+
+		newSelected = installedLanguages[ installedIndex + 1 ];
+
+		if (
+			! newSelected &&
+			installedLanguages[ 0 ] !== selectedInactiveLanguage
+		) {
+			newSelected = installedLanguages[ 0 ];
+		}
+
+		if ( ! newSelected ) {
+			newSelected = availableLanguages[ availableIndex + 1 ];
+
+			if ( availableLanguages[ 0 ] !== selectedInactiveLanguage ) {
+				newSelected = availableLanguages[ 0 ];
+			}
+		}
+
+		setSelectedInactiveLanguage( newSelected );
+
+		speak( __( 'Locale added to list' ) );
+	};
+
+	const onRemove = () => {
+		const foundIndex = languages.findIndex(
+			( { locale } ) => locale === selectedLanguage?.locale
+		);
+
+		setSelectedLanguage(
+			languages[ foundIndex + 1 ] || languages[ foundIndex - 1 ]
+		);
+
+		setLanguages( ( prevLanguages ) =>
+			prevLanguages.filter(
+				( { locale } ) => locale !== selectedLanguage?.locale
+			)
+		);
+		_setLanguages( ( prevLanguages ) =>
+			prevLanguages.filter(
+				( { locale } ) => locale !== selectedLanguage?.locale
+			)
+		);
+
+		speak( __( 'Locale removed from list' ) );
+
+		if ( languages.length === 1 ) {
+			let emptyMessageA11y = sprintf(
+				/* translators: %s: English (United States) */
+				__( 'No languages selected. Falling back to %s.' ),
+				'English (United States)'
+			);
+
+			if ( showOptionSiteDefault ) {
+				emptyMessageA11y = __(
+					'No languages selected. Falling back to Site Default.'
+				);
+			}
+
+			speak( emptyMessageA11y );
+		}
+	};
+
+	const onMoveUp = () => {
+		setLanguages( ( prevLanguages ) => {
+			const srcIndex = prevLanguages.findIndex(
+				( { locale } ) => locale === selectedLanguage?.locale
+			);
+			return reorder(
+				Array.from( prevLanguages ),
+				srcIndex,
+				srcIndex - 1
+			);
+		} );
+
+		speak( __( 'Locale moved up' ) );
+	};
+
+	const onMoveDown = () => {
+		setLanguages( ( prevLanguages ) => {
+			const srcIndex = prevLanguages.findIndex(
+				( { locale } ) => locale === selectedLanguage?.locale
+			);
+			return reorder< Language[] >(
+				Array.from( prevLanguages ),
+				srcIndex,
+				srcIndex + 1
+			);
+		} );
+
+		speak( __( 'Locale moved down' ) );
+	};
+
+	const onKeyDown = ( event: KeyboardEvent< HTMLElement > ) => {
```

Comment:
Looks like I accidentally removed this when I ported the component to Gutenberg 😄  I'll update it in a bit.

The way it's supposed to work is this:

* ArrowUp: Select the locale above
* Alt+ArrowUp: Move the current locale one up

Reply:
Cool, thanks for confirming 👍 

What a bummer that Alt+ArrowUp will conflict with Storybook's navigation controls, and you'll navigate to the previous story in the navigation menu. I assume Alt+ArrowDown will conflict the next story shortcut, but there's no next story in the nav menu to test 😅 Anything we can do to prevent the shortcuts from bumping into each other?

Reply:
Are the ALT+arrow standard patterns for this kind of interaction? 

> conflict with Storybook's navigation controls

That's a pity, but I'm afraid that attempting to improve this aspect could actually cause us to design worse APIs

Reply:
Is it something we can disable per story in the storybook config, perhaps?

Reply:
Not sure if we can selectively disable only those shortcuts. We may have to otherwise add a `onKeyDown` listener to the wrapper element, and `preventDefault` / `stopPropagation` for the specific story

Reply:
> Are the ALT+arrow standard patterns for this kind of interaction?

AFAIK yes. The shortcuts were originally discussed in detail with the WP core accessibility team.

> Is it something we can disable per story in the storybook config, perhaps?

There's an `enableShortcuts` option, see https://storybook.js.org/docs/configure/user-interface/features-and-behavior

Reply:
> There's an `enableShortcuts` option

Yup, but it looks like that would disabled all shortcuts, which IMO is not something we'd like to do

Reply:
Introducing a wrapper element and using it to capture `keyDown` events as @ciampo suggested sounds like the best compromise:

> We may have to otherwise add a `onKeyDown` listener to the wrapper element, and `preventDefault` / `stopPropagation` for the specific story

We can even add a little disclaimer at the bottom of the story that the Storybook prev/next story control is disabled or something.

Reply:
Storybook shortcuts are now disabled and there is a note in the story
<img width="1082" alt="Screenshot 2024-09-13 at 15 12 16" src="https://github.com/user-attachments/assets/645d1cad-4ffd-4ed4-944f-09237425db00">